diffkemp/diffkemp.py

```diff
@@ -580,3 +587,164 @@ def text_indent(text, width):
             if not output_dir:
                 output.write(text_indent("}}}\n", indent - 2))
             output.write("\n")
+
+
+class YamlOutput:
+    """Yaml representation of compare result."""
+    def __init__(self, snapshot_dir_old, snapshot_dir_new, result):
+        self.old_dir = os.path.join(
+            os.path.abspath(snapshot_dir_old), "")
+        self.new_dir = os.path.join(
+            os.path.abspath(snapshot_dir_new), "")
+        self.result = result
+        self.output = {}
+
+        self._create_output()
+
+    def save(self, output_dir, file_name):
+        with open(os.path.join(output_dir, file_name), "w") as file:
+            yaml.dump(self.output, file, sort_keys=False)
+
+    def _create_output(self):
+        """Creates yaml representation of compare result."""
+        self.output["old-snapshot"] = self.old_dir
+        self.output["new-snapshot"] = self.new_dir
+
+        # Sets with symbol names
+        function_names = set()
+        macro_names = set()
+        # Note: type_names contains tuples (type name, parent name)
+        type_names = set()
+
+        self.output["results"] = self._get_results(function_names, type_names,
+                                                   macro_names)
+        self.output["definitions"] = self._get_definitions(function_names,
+                                                           type_names,
+                                                           macro_names)
+
+    def _get_results(self, function_names, type_names, macro_names):
+        """Returns call stacks of not-equal functions.
+        Parameters are output arguments to save names of symbols that appear
+        in call stacks.
+        """
+        # list of all not-equal functions callstacks
+        results = []
+        for fun_name, fun_result in self.result.inner.items():
+            if fun_result.kind == Result.Kind.NOT_EQUAL:
+                function_names.add(fun_name)
+                # list of not-equal functions callstacks for a function
+                diffs = []
+                for called_res in sorted(fun_result.inner.values(),
+                                         key=lambda r: r.first.name):
+                    # information about a function which is different
+                    diff = {}
+                    diff["function"] = called_res.first.name
+                    # getting callstacks
+                    for entity in [called_res.first, called_res.second]:
```

Comment:
Could you extract the below code into a function (something like `_get_callstack`) and then do
```python
diff["callstack-old"] = self._get_callstack(...)
diff["callstack-new"] = self._get_callstack(...)
```
Maybe if you create a separate `Callstack` class (as mentioned elsewhere), it could be a part of that (something like `Callstack.to_yaml()`).

Reply:
Thanks, I have created the `Callstack` class and moved getting of call stack to method `to_output_yaml_with_rel_path` and getting of "symbols" (from call stack for getting definitions) to method `get_symbol_names`.